core/services/vrf/listener_v2.go

```diff
@@ -735,6 +743,14 @@ func (lsn *listenerV2) processRequestsPerSub(
 		}
 		maxGasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddress)
 
+		// Cases:
+		// 1. Never simulated: in this case, we want to observe the time until simulated
+		// on the utcTimestamp field of the pending request.
+		// 2. Simulated before: in this case, lastTry will be set to a non-zero time value,
+		// in which case we'd want to use that as a relative point from when we last tried
+		// the request.
```

Comment:
Do we expect both cases to always have similar duration? I wonder if it would be easier to reason about these metrics if we separated them into 2 different metrics (`vrf_request_time_until_initial_sim`,  `vrf_sim_retry_delay`). 

For instance, if case #2 has higher average duration than case #1 and customer request pattern changes (e.g. high load), the combined metric will be skewed more towards case #2.

Reply:
I did think about that, and I thought it might be simpler to combine it into the same metric, but I think you're right in that we would see a skew in case (2) that would be hard to explain only with this metric.

Another issue with this metric as it is now is that it measures wall clock time as opposed to # of blocks, whereas number of blocks would be more meaningful, especially on slower chains.

Reply:
Addressed - made them two separate metrics.